io/FV3GFS_io.F90

```diff
@@ -2127,9 +2249,34 @@ subroutine sfc_prop_restart_write (Sfcprop, Atm_block, Model, fv_domain, timesta
         sfc_name2(nvar2m+45) = 'smcwtdxy'
         sfc_name2(nvar2m+46) = 'deeprechxy'
         sfc_name2(nvar2m+47) = 'rechxy'
+        nvar2me = nvar2m+47
       endif
    end if
 
+   if(Model%lkm>0) then
+     if(Model%iopt_lake==Model%iopt_lake_flake  ) then
+       if(Model%me==0) then
+         if(size(sfc_name2)/=nvar2me+10) then
+3814       format("ERROR: size mismatch size(sfc_name2)=",I0," /= nvar2me+10=",I0)
+           write(0,3814) size(sfc_name2),nvar2me+10
+         endif
+       endif
+       sfc_name2(nvar2me+1) = 'T_snow'
+       sfc_name2(nvar2me+2) = 'T_ice'
+       sfc_name2(nvar2me+3)  = 'h_ML'
+       sfc_name2(nvar2me+4)  = 't_ML'
+       sfc_name2(nvar2me+5)  = 't_mnw'
+       sfc_name2(nvar2me+6)  = 'h_talb'
+       sfc_name2(nvar2me+7)  = 't_talb'
+       sfc_name2(nvar2me+8)  = 't_bot1'
+       sfc_name2(nvar2me+9)  = 't_bot2'
+       sfc_name2(nvar2me+10)  = 'c_t'
+     else if(Model%iopt_lake==Model%iopt_lake_clm) then
+       ! Tell clm_lake to register all of its fields
+       call clm_lake%register_fields(Sfc_restart)
+     endif
+   endif
+   
    !--- register the 2D fields
```

Comment:
Add the following section after line 2321. Chunxi has successfully run control_flake_restart with this revision.
!Flake
   if (Model%lkm > 0 .and. Model%iopt_lake==Model%iopt_lake_flake) then
        nvar_before_lake=nvar2m+nvar2o+nvar2r+nvar2mp
      mand = .false.
        do num = nvar_before_lake+1,nvar_before_lake+nvar2l
         var2_p => sfc_var2(:,:,num)
         call register_restart_field(Sfc_restart, sfc_name2(num),var2_p,dimensions=(/'xaxis_1', 'yaxis_1', 'Time   '/),&
                                     &is_optional=.not.mand)
       enddo
   endif

Reply:
Although the flake restart does not crash, it also does not reproduce the baseline. The 2threads, decomp, and restart tests all do not match the baseline. This test was with hera.gnu. For your convenience, I've added a `tests/flake.conf` with all of the flake tests. You'll need to generate baselines (rt.sh -c) and compare against them (rt.sh -m).

Reply:
Yes. The results are quite different. I am debugging. Did you run any clm lake restart case? Does it reproduce?

Reply:
I'm putting together an rrfs.conf file with all of the relevant tests so you can use the target model, instead of gfs v16. It'll have flake, clm lake, and no lake, with debug & non-debug, and the other usual test variants. It'll take a few hours for me to test it.